src/app/desktop/utils/downloader.js

```diff
@@ -11,7 +11,8 @@ const fs = fss.promises;
 export const downloadInstanceFiles = async (
   arr,
   updatePercentage,
-  threads = 4
+  threads = 4,
+  InstancePath
```

Comment:
This is a no-no. The complete path should be built before passing the array to this function, not after...